diff --git a/rosidl_adapter/resource/__init__.py b/rosidl_adapter/resource/__init__.py
index ce33358..427cac2 100644
--- a/rosidl_adapter/resource/__init__.py
+++ b/rosidl_adapter/resource/__init__.py
@@ -44,19 +44,24 @@ def evaluate_template(template_name, data):
     template_path = os.path.join(os.path.dirname(__file__), template_name)
 
     output = StringIO()
+    config = em.Configuration()
+    config.deleteOnError = True
+    config.rawErrors = True
     try:
         _interpreter = em.Interpreter(
             output=output,
-            options={
-                em.BUFFERED_OPT: True,
-                em.RAW_OPT: True,
-            })
+            config=config,
+#             options={
+#                 em.BUFFERED_OPT: True,
+#                 em.RAW_OPT: True,
+#             }
+        )
 
         with open(template_path, 'r') as h:
             content = h.read()
         _interpreter.invoke(
             'beforeFile', name=template_name, file=h, locals=data)
-        _interpreter.string(content, template_path, locals=data)
+        _interpreter.string(content, locals=data)
         _interpreter.invoke('afterFile')
 
         return output.getvalue()
@@ -78,7 +83,7 @@ def _evaluate_template(template_name, **kwargs):
             'beforeInclude', name=template_path, file=h, locals=kwargs)
         content = h.read()
     try:
-        _interpreter.string(content, template_path, kwargs)
+        _interpreter.string(content, kwargs)
     except Exception as e:  # noqa: F841
         print(
             f"{e.__class__.__name__} processing template '{template_name}': "
